src/utils/viewer/viewer.cpp

```diff
@@ -820,33 +705,89 @@ void Viewer::keyPressEvent(KeyEvent& event) {
       torqueLastObject();
       break;
     case KeyEvent::Key::N:
-      toggleNavMeshVisualization();
+      // toggle navmesh visualization
+      simulator_->setNavMeshVisualization(
+          !simulator_->isNavMeshVisualizationActive());
       break;
     case KeyEvent::Key::I:
       Mn::DebugTools::screenshot(Mn::GL::defaultFramebuffer,
                                  "test_image_save.png");
       break;
+    case KeyEvent::Key::Q:
+      // query the agent state
+      logAgentStateMsg(true, true);
+      break;
     case KeyEvent::Key::B: {
       // toggle bounding box on objects
       drawObjectBBs = !drawObjectBBs;
-      for (auto id : physicsManager_->getExistingObjectIDs()) {
-        physicsManager_->setObjectBBDraw(id, &sceneGraph_->getDrawables(),
-                                         drawObjectBBs);
+      for (auto id : simulator_->getExistingObjectIDs()) {
+        simulator_->setObjectBBDraw(drawObjectBBs, id);
       }
     } break;
+    case KeyEvent::Key::H:
+      printHelpText();
+      break;
     default:
       break;
   }
-  if (agentMoved) {
-    logAgentStateMsg(true, true);
-  }
-  updateRenderCamera();
   redraw();
 }
 
-void Viewer::updateRenderCamera() {
-  renderCamera_->node().setTransformation(
-      rgbSensorNode_->absoluteTransformation());
+void Viewer::printHelpText() {
+  Cr::Utility::Debug() << "==================================================";
+  Cr::Utility::Debug() << "Welcome to the Habitat-sim C++ Viewer application!";
+  Cr::Utility::Debug() << "==================================================";
+  Cr::Utility::Debug() << " Mouse Functions:";
+  Cr::Utility::Debug() << " ----------------";
+  Cr::Utility::Debug() << "   LEFT: ";
+  Cr::Utility::Debug()
+      << "     Click and drag to rotate the agent and look up/down.";
+  Cr::Utility::Debug() << "   RIGHT: ";
+  Cr::Utility::Debug()
```

Comment:
I'd instead put this into a huge `R"()"` literal and print that, it's less work and easier to update ;)

It could also be put to Utility::Arguments via `setGlobalHelp()` (as the current text there is rather useless) ... but that's shown only if the user explicitly asks for `--help` so that's probably not so useful.